File: src/main/java/com/kamegatze/learnjvm/utils/impl/HtmlConvectorImpl.java
package com.kamegatze.learnjvm.utils.impl;

import com.kamegatze.learnjvm.utils.HtmlConvector;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.springframework.stereotype.Component;

import java.util.Objects;

@Component
public class HtmlConvectorImpl implements HtmlConvector {
    @Override
    public String toFirstAttributeToText(String nameAttribute, String content) {
        Document document = Jsoup.parse(content);
        Element element = document.selectFirst(nameAttribute);
        return Objects.isNull(element) ? "" : element.text();
    }
}

File: src/main/kotlin/com/kamegatze/learnjvm/utils/impl/HtmlConvectorImpl.kt
package com.kamegatze.learnjvm.utils.impl

import com.kamegatze.learnjvm.utils.HtmlConvector
import org.jsoup.Jsoup
import org.springframework.stereotype.Component

@Component
class HtmlConvectorImpl : HtmlConvector {
    override fun toFirstAttributeToText(nameAttribute: String, content: String): String {
        val document = Jsoup.parse(content)
        return document.selectFirst(nameAttribute)?.text() ?: ""
    }
}